4/tests/unit/GraphQLBackendWithNeo4J.spec.js

```diff
@@ -0,0 +1,271 @@
+import {
+    FILTER_MODE,
+    TodoNeo4JAPI,
+} from '../../apollo-server/utils/TodoNeo4JAPI'
+
+import { createJwt } from '../../apollo-server/utils/jwtCreator'
+import fs from 'fs'
+import path from 'path'
+import resolvers from '../../apollo-server/resolvers'
+
+var neo4j = require('neo4j-driver')
+const { ApolloServer, gql } = require('apollo-server')
+const { createTestClient } = require('apollo-server-testing')
+
+const typeDefs = fs.readFileSync(
+    path.resolve(__dirname, '../../apollo-server/schema.graphql'),
+    { encoding: 'utf8' }
+)
+
+const ADD_TODO = gql`
+    mutation addTodo($id: Int!, $newMessage: String!, $userAuth: String!) {
+        addTodo(id: $id, newMessage: $newMessage, userAuth: $userAuth) {
+            id
+            message
+        }
+    }
+`
+
+const UPDATE_TODO = gql`
+    mutation updateTodo($id: Int!, $updateMessage: String!) {
+        updateTodo(id: $id, updateMessage: $updateMessage)
+    }
+`
+
+const ADD_DEPENDENCY = gql`
+    mutation addToDoDependency($id: Int!, $dependencyId: Int!) {
+        addToDoDependency(id: $id, dependencyId: $dependencyId)
+    }
+`
+
+const GET_DEPENDENCIES = gql`
+    query getToDoDependencies($id: Int!) {
+        getToDoDependencies(id: $id){
+            id
+            message
+        }
+    }
+`
+
+const DELETE_TODO = gql`
+    mutation deleteTodo($id: Int!) {
+        deleteTodo(id: $id)
+    }
+`
+
+const GET_TODO = gql`
+    query todo($id: Int!) {
+        todo(id: $id) {
+            id
+            message
+        }
+    }
+`
+
+const GET_TODOS = gql`
+    query todos($FILTER_MODE: String!, $skip: Int!, $limit: Int!) {
+        todos(FILTER_MODE: $FILTER_MODE, skip: $skip, limit: $limit) {
+            id
+            message
+        }
+    }
+`
+
+// const GET_TODOS_FOR_USER = gql`
+//   query todosForUser($userAuth: String!) {
+//     todosForUser(userAuth: $userAuth) {
+//       id
+//       message
+//     }
+//   }
+// `;
+
+function createNewDriver() {
+    return new neo4j.driver(
+        'bolt://localhost:7687',
+        neo4j.auth.basic('neo4j', 'ggs')
+    )
+}
+
+function createNewServer(driver) {
+    return new ApolloServer({
+        typeDefs,
+        resolvers,
+        dataSources: () => ({
+            ds: new TodoNeo4JAPI(driver),
+        }),
+        mockEntireSchema: false,
+        formatError: err => {
+            console.log(err)
+            return err
+        },
+    })
+}
+
+async function addMultipleTodos(times, testClientQuery, jwtToken) {
+    for (let i = 0; i < times; i++) {
```

Comment:
Don't use C-style for loops:
```suggestion
    for (i of times) {
```